src/Passage.sol

```diff
@@ -56,6 +61,45 @@ contract Passage {
         emit Enter(rollupChainId, address(0), rollupRecipient, msg.value);
     }
 
+    /// @dev See `enter` above for docs.
+    function enter(address rollupRecipient) external payable {
+        enter(defaultRollupChainId, rollupRecipient);
+    }
+
+    /// @notice Allows a special transaction to be sent to the rollup with sender == L1 msg.sender.
+    /// @dev Transaction is processed after normal rollup block execution.
+    /// @param rollupChainId - The rollup chain to send the transaction to.
+    /// @param to - The address to call on the rollup.
+    /// @param data - The data to send to the rollup.
+    /// @param value - The amount of Ether to send on the rollup.
+    /// @param gas - The gas limit for the transaction.
+    /// @param maxPrioFee - The maximum priority fee for the transaction.
+    /// @param maxBaseFee - The maximum base fee for the transaction.
+    /// @custom:emits Transact indicating the transaction to mine on the rollup.
+    function transact(
+        uint256 rollupChainId,
+        address to,
+        bytes calldata data,
+        uint256 value,
+        uint256 gas,
+        uint256 maxPrioFee,
+        uint256 maxBaseFee
+    ) public {
```

Comment:
we should probably make payable variants too, that call enter and then transact

Reply:
Enter with recipient == msg.sender always, or add another param for Ether recipient?

```
    function enterAndTransact(
        uint256 rollupChainId,
        address etherRecipient,
        address to,
        bytes calldata data,
        uint256 value,
        uint256 gas,
        uint256 maxPrioFee,
        uint256 maxBaseFee,
    ) external payable {
        enter(rollupChainId, etherRecipient);
        transact(rollupChainId, to, data, value, gas, maxPrioFee, maxBaseFee);
    }

    function enterAndTransact(
        uint256 rollupChainId, 
        address to,
        bytes calldata data,
        uint256 value,
        uint256 gas,
        uint256 maxPrioFee,
        uint256 maxBaseFee,
    ) external payable {
        enter(rollupChainId, msg.sender);
        transact(rollupChainId, to, data, value, gas, maxPrioFee, maxBaseFee);
    }
```

Reply:
> Enter with recipient == msg.sender always, or add another param for Ether recipient?

add another param I think 🤔

Reply:
hmmmm interesting. that's def an option.

thinking about this, it feels like the value of adding `enter` to `transact` is that if the sender doesn't have ETH to pay gas, they can provide themselves ETH to pay gas + send a transaction. here's an idea which is optimized for that flow: https://github.com/init4tech/zenith/pull/45/commits/0c46e69b7412fe00e14df0dfd000c04d3bdd5d19

trying to think of a flow where it would make sense to want to send ether to a different address on the RU and then transact. it's def possible but probably significantly less likely case, which could pretty much be served just as well by doing `enter` + `transact` via separate fn calls. 

gonna mull on this for a sec.

Reply:
> trying to think of a flow where it would make sense to want to send ether to a different address on the RU and then transact. it's def possible but probably significantly less likely case, which could pretty much be served just as well by doing enter + transact via separate fn calls.

I guess the question is whether there's any user that would want to enter to another address atomically with the transact. like the enter happening first matters to them. I don't want to rule out that use case. but it's definitely not the one we're optimizing for

Reply:
definitely. a smart contract could of course just could call enter separately. an EOA that wants to enter to a diff address + call transact atomically guaranteed within one block would be the only case that's not really served by having separate interfaces

Reply:
adding an argument doesn't cost much i just hate function signatures with a ton of arguments that you have to grok

Reply:
end of the day maybe i'll just add the argument. sigh

Reply:
yeah, i agree it's a bit of an unwieldy api 🤔  should the args be one struct maybe?

Reply:
honestly i think that as a solution for too many args is probably worse

Reply:
https://github.com/init4tech/zenith/pull/45/commits/624a7e81e008b30ed7161a5dada47bc1d31a7bea